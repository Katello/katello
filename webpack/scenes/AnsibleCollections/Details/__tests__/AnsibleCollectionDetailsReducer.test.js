import { testReducerSnapshotWithFixtures } from 'react-redux-test-utils';
import reducer from '../AnsibleCollectionDetailsReducer';
import {
  ANSIBLE_COLLECTION_DETAILS_REQUEST,
  ANSIBLE_COLLECTION_DETAILS_SUCCESS,
  ANSIBLE_COLLECTION_DETAILS_ERROR,
} from '../AnsibleCollectionDetailsConstants';

const fixtures = {
  'should return the initial state': {},
  'should handle ANSIBLE_COLLECTION_DETAILS_REQUEST': {
    action: {
      type: ANSIBLE_COLLECTION_DETAILS_REQUEST,
    },
  },
  'should handle ANSIBLE_COLLECTION_DETAILS_SUCCESS': {
    action: {
      type: ANSIBLE_COLLECTION_DETAILS_SUCCESS,
      payload: {
        results: { data: { id: 1 } },
      },
    },
  },
  'should handle ANSIBLE_COLLECTION_DETAILS_ERROR': {
    action: {
      type: ANSIBLE_COLLECTION_DETAILS_ERROR,
      payload: new Error('things have gone terribly wrong'),
    },
  },
};

describe('AnsibleCollectionDetails reducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
