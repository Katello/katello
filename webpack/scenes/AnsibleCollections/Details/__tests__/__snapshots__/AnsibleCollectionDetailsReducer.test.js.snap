// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnsibleCollectionDetails reducer should handle ANSIBLE_COLLECTION_DETAILS_ERROR 1`] = `
Object {
  "error": "things have gone terribly wrong",
  "itemCount": 0,
  "loading": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "results": Array [],
}
`;

exports[`AnsibleCollectionDetails reducer should handle ANSIBLE_COLLECTION_DETAILS_REQUEST 1`] = `
Object {
  "itemCount": 0,
  "loading": true,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "results": Array [],
}
`;

exports[`AnsibleCollectionDetails reducer should handle ANSIBLE_COLLECTION_DETAILS_SUCCESS 1`] = `
Object {
  "itemCount": 0,
  "loading": false,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "results": Array [],
}
`;

exports[`AnsibleCollectionDetails reducer should return the initial state 1`] = `
Object {
  "itemCount": 0,
  "loading": true,
  "pagination": Object {
    "page": 0,
    "perPage": 20,
  },
  "results": Array [],
}
`;
